pkg/secret/secret_test.go

```diff
@@ -0,0 +1,174 @@
+package secret
+
+import (
+	"github.com/stretchr/testify/assert"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/kubernetes/fake"
+	"testing"
+)
+
+type HelperKubernetes struct{}
+
+func (k *HelperKubernetes) ResourceContains(namespace, value string, resource schema.GroupVersionResource) (bool, error) {
+	if "nameA" == value {
+		return false, nil
+	} else {
+		return true, nil
+	}
+}
+
+var now = metav1.Now()
+var testNamespace = "testNamespace"
+
+func Test_ListNamesAndLabels(t *testing.T) {
+	tests := []struct {
+		name        string
+		secrets     []v1.Secret
+		secretNames []string
+		labels      []string
+		err         error
+	}{
+		{
+			name:        "GivenListOfSecrets_WhenLabelsAndNamesDefined_ThenReturnNamesAndLabels",
+			secrets:     generateBaseTestSecrets(),
+			secretNames: []string{"nameA", "nameB"},
+			labels:      []string{"keyA=valueA", "keyB=valueB", "keyC=valueC"},
+		},
+		{
+			name: "GivenListOfSecrets_WhenOnlyNamesDefined_ThenReturnNamesWithNoLabels",
+			secrets: []v1.Secret{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:              "nameA",
+						Namespace:         testNamespace,
+						CreationTimestamp: now,
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:              "nameB",
+						Namespace:         testNamespace,
+						CreationTimestamp: now,
+					},
+				},
+			},
+			secretNames: []string{"nameA", "nameB"},
+			labels:      []string{},
```

Comment:
Please add more test cases:
* where labels are used/expected
* Check for errors

Reply:
* the first test covers the labels as well
* added check on errors on a previous conversation

Reply:
added